templates/azml-app/Dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM mcr.microsoft.com/azureml/minimal-py312-inference:latest
+
+USER root
+
+RUN apt-get update && apt-get install -y --no-install-recommends curl
+
+# Copy azcopy installation script and run it.
+COPY azcopy-installer.sh /azcopy-installer.sh
+RUN chmod +x /azcopy-installer.sh
+RUN /azcopy-installer.sh

```

Comment:
nit: can combine all of these into one RUN command to minimize layers and improve caching

RUN apt-get update && \
    apt-get install -y --no-install-recommends curl && \
    chmod +x /azcopy-installer.sh && \
    /azcopy-installer.sh && \
    rm -rf /var/lib/apt/lists/*